### PR TITLE
lib: modem_key_mgmt: update modem_key_mgmt.c

### PR DESCRIPTION
**Pull request by MechanicV1**
Added support to return 1 when buffer length doesnot match certificate length

### CODE REVIEW

File: lib/modem_key_mgmt/modem_key_mgmt.c [hunk 1]
```diff
@@ -199,16 +199,25 @@ int modem_key_mgmt_cmp(nrf_sec_tag_t sec_tag,
...
+	}
+
+	end = strchr(begin, '\"');
+	if (!end) {
+		return -ENOENT;
+	}
+
+	if (end - begin != len) {
+		return -ENOMEM;
 	}
```
Comment (lemrey) [inline lib/modem_key_mgmt/modem_key_mgmt.c:218]: The check is correct, but the error is wrong. This function return 0 on match, 1 on mismatch (similar to `memcmp` semantics), or a negative value on errors. So in this case, where the length of the credential in modem does not match the length of the credential the application is checking against, we should return 1.

**Review by eivindj-nordic — APPROVED** [review]
LGTM, I think we need a changelog entry, though.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -284,7 +284,10 @@ Debug libraries
 Modem libraries
 ---------------
 
-|no_changes_yet_note|
+* Modem key Management library:
+
+   * Updated :c:func:`modem_key_mgmt_cmp` function to return 1 if buffer length does not match the certificate length.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:289]:
```suggestion
* :ref:`nrf_modem_lib_readme`:
  
  * Updated the :c:func:`modem_key_mgmt_cmp` function to return ``1`` if the buffer length does not match the certificate length.
```

**Review by lemrey — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Comment by cvinayak** [comment]
[12:58] Rathakrishnan, Nirmal
    looks like github is not picking the latest run again, https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-11457/
​[12:58] Rathakrishnan, Nirmal
    its good to merge Chettimada, Vinayak

**Pull request merged**